packages/tao/src/shared/workspace.ts

```diff
@@ -23,39 +22,8 @@ export interface WorkspaceJsonConfiguration {
   /**
    * Projects' configurations
    */
-  projects: { [projectName: string]: ProjectConfiguration };
-
-  /**
-   * Default project. When project isn't provided, the default project
-   * will be used. Convenient for small workspaces with one main application.
-   */
-  defaultProject?: string;
-
-  /**
-   * List of default values used by generators.
-   *
-   * These defaults are global. They are used when no other defaults are configured.
-   *
-   * Example:
-   *
-   * ```
-   * {
-   *   "@nrwl/react": {
-   *     "library": {
-   *       "style": "scss"
-   *     }
-   *   }
-   * }
-   * ```
-   */
-  generators?: { [collectionName: string]: { [generatorName: string]: any } };
-
-  /**
-   * Default generator collection. It is used when no collection is provided.
-   */
-  cli?: {
-    packageManager?: PackageManager;
-    defaultCollection?: string;
+  projects: {
+    [projectName: string]: ProjectConfiguration & NxJsonProjectConfiguration;
```

Comment:
Yeah, removing this is technically breaking as it could break community plugins that manually read these fields. 

I went ahead and removed all references to NxJsonProjectConfiguration across the codebase, and deprecated the type. Do you think we should remove it without deprecating? I can go ahead w/ that if its the preference.